test/merkle/merkle.cpp

```diff
@@ -77,6 +91,18 @@ BOOST_AUTO_TEST_CASE(merkletree_validate_test_2) {
     testing_validate_template<hashes::blake2b<224>, 3>(v);
 }
 
+BOOST_AUTO_TEST_CASE(merkletree_validate_test_3) {
+    using hash_type = hashes::pedersen<>;
+    std::size_t leaf_number = 8;
+    std::vector<std::array<bool, hash_type::digest_bits>> v;
+    for (std::size_t i = 0; i < leaf_number; ++i) {
+        std::array<bool, hash_type::digest_bits> leaf;
+        std::generate(leaf.begin(), leaf.end(), [&]() { return std::rand() % 2; });
+        v.emplace_back(leaf);
+    }
+    testing_validate_template_pedersen<hashes::pedersen<>, 2>(v);
```

Comment:
It's good test for testing tree with random numbers. Can we rename it in something like `testing_validate_template_random_data` and also call for another hashes like in `merkletree_validate_test_2`?